### PR TITLE
DVDInterface: Make changing discs savestate-safe

### PR DESCRIPTION
**Pull request by JosJuice**
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/4242)

### CODE REVIEW

File: Source/Core/Core/HW/DVDInterface.cpp [hunk 1]
```diff
@@ -244,6 +244,8 @@ static CoreTiming::EventType* s_dtk;
 static u64 s_last_read_offset;
 static u64 s_last_read_time;
 
+std::string s_disc_path_to_insert;
```
Comment (leoetlino) [inline Source/Core/Core/HW/DVDInterface.cpp]: I'm not sure, but shouldn't this be `static`?

Comment (JosJuice) [inline Source/Core/Core/HW/DVDInterface.cpp]: Fixed.

**Comment by degasus** [comment]
LGTM

**Comment by JosJuice** [comment]
I added another commit. @degasus, can you re-review?

**Comment by degasus** [comment]
Yeah, is still fine. But you need a rebase now.

**Comment by JosJuice** [comment]
Rebased.

**Pull request merged**